File: components/credentials/src/main/java/org/cloudfoundry/credhub/credential/StringCredentialValue.java
package org.cloudfoundry.credhub.credential;

import java.util.Objects;

import javax.validation.constraints.NotEmpty;

import com.fasterxml.jackson.annotation.JsonValue;
import org.cloudfoundry.credhub.ErrorMessages;

public class StringCredentialValue implements CredentialValue {

  @NotEmpty(message = ErrorMessages.MISSING_VALUE)
  private final String string;

  public StringCredentialValue(final String password) {
    super();
    this.string = password;
  }

  @JsonValue
  public String getStringCredential() {
    return string;
  }

  @Override
  public boolean equals(final Object o) {
    if (this == o) {
      return true;
    }
    if (o == null || getClass() != o.getClass()) {
      return false;
    }
    final StringCredentialValue that = (StringCredentialValue) o;
    return Objects.equals(string, that.string);
  }

  @Override
  public int hashCode() {
    return Objects.hash(string);
  }
}

File: components/credentials/src/main/kotlin/org/cloudfoundry/credhub/credential/StringCredentialValue.kt
package org.cloudfoundry.credhub.credential

import com.fasterxml.jackson.annotation.JsonValue
import org.cloudfoundry.credhub.ErrorMessages
import java.util.Objects
import javax.validation.constraints.NotEmpty

class StringCredentialValue(
    @field:NotEmpty(message = ErrorMessages.MISSING_VALUE)
    @get:JsonValue
    val stringCredential: String,
) : CredentialValue {
    override fun equals(other: Any?): Boolean {
        if (this === other) {
            return true
        }
        if (other == null || javaClass != other.javaClass) {
            return false
        }
        val that = other as StringCredentialValue?
        return stringCredential == that!!.stringCredential
    }

    override fun hashCode(): Int = Objects.hash(stringCredential)
}
